### PR TITLE
[node] Expose global ResponseInit; remove NodeJS.fetch namespace

### PR DESCRIPTION
**Pull request by thw0rted**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://fetch.spec.whatwg.org/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by thw0rted** [comment]
This is really two changes and if anyone objects to either, I can split into separate PRs.

The first is to add the global `ResponseInfo` type, which was missing from the original PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66824.

The second is to refactor out the `NodeJS.fetch` "fictional" namespace so that users don't start explicitly including the types there, including aliases that are only used as part of the conditional global definition gadget (`typeof globalThis extends ...`).

I updated tests but couldn't get `test-all` to run locally (https://github.com/microsoft/DefinitelyTyped-tools/issues/816) and couldn't figure out how to get it to test the `v18` subdirectory.  Hopefully the CI tests cover both those cases?

**Comment by thw0rted** [comment]
I'm not sure these CI failures are actually caused by my changes, see also https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67259 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67291.

Maybe @jakebailey or @andrewbranch can take a look?  (Author and reviewer of https://github.com/microsoft/DefinitelyTyped-tools/pull/803, respectively.)

**Comment by thw0rted** [comment]
New commit adds the missing [global File type](https://nodejs.org/api/globals.html#class-file) as well, since it winds up in some fetch sub-sub-types.

**Comment by jakebailey** [comment]
Weird es-abstract related failures are a problem due to our new partial package installs here; the latest dt-tools should fix this via some hack.

**Comment by jakebailey** [comment]
wat 

image

File: types/node/test/globals.ts
Comment (RyanCavanaugh) [inline types/node/test/globals.ts:89]:
```suggestion
    f.name; // $ExpectType string
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/globals.d.ts
1. types/node/v18/globals.d.ts
1. types/node/ts4.8/globals.d.ts
1. types/node/v18/ts4.8/globals.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3884171509ea897c7411be2f8d23f81dd99c1c62
</p>

File: types/node/test/globals.ts
Comment (jakebailey) [reply types/node/test/globals.ts]: FWIW this would be fixed if you run `dprint fmt` 😄

File: types/node/globals.d.ts
Comment (andrewbranch) [inline types/node/globals.d.ts:1]: This makes `import {} from "node/globals"` legal 😕 That’s why my `_Request` et al. types are accessible. There’s no way to hide them without declaring a module.

Comment (thw0rted) [reply types/node/globals.d.ts:1]: I guess you learn something new every day.  When I [messed this up last time](https://github.com/thw0rted/DefinitelyTyped/blob/node-add-responseinit/types/node/dom-events.d.ts), nobody seems to have noticed.  Does that need to change as well?

Would it be preferable to move these declarations out of `globals.d.ts` and into some other module context?  I was mostly trying to avoid exposing the fictional namespace so that library authors wouldn't come to depend on that implementation detail, but of course I'd defer to the team about which hazard is more important to work around.

Comment (andrewbranch) [reply types/node/globals.d.ts:1]: Well, if nobody noticed the last one, probably nobody will notice this. I guess it’s fine 😄 

If we had a type reference directive condition, we could use it to hide everything from imports while keeping things accessible to type reference directives:


```json
{
  "exports": {
    "./*": {
      "types@>=5.5": {
        "typereference": "./*",
        "default": null
      },
      "default": "./*"
    }
  }
}
```

🤔

**Review by andrewbranch — APPROVED** [review]

**Comment by thw0rted** [comment]
Ready to merge

**Comment by jordanebelanger** [comment]
Hi folks,

I am having an issue which may stem from this PR. Before this PR, `RequestCredentials` and other types were exported directly.

Now I am trying to make the OpenAPi typescript-fetch code generator work for nodejs and running into an issue whereas it's referencing types such as `RequestCredentials` directly. This no longer works following this PR as the type is no longer exported directly. 

But, this is something that does work in a context where lib.dom is added as a lib.

image

**Comment by thw0rted** [comment]
You can try an older version of `@types/node` but I don't think it ever exposed a global `RequestCredentials` -- this PR only removed a namespaced re-export of `import("undici-types").RequestCredentials`.  There are a number of helper types (interfaces and enums, mostly) that `lib.dom.d.ts` exports but which have no runtime value, and this is one of them.  You could open a PR to add the helpers listed [here in the fetch spec](https://fetch.spec.whatwg.org/#idl-index), for compatibility with lib-DOM, if you wanted.  I'm not a maintainer, but it seems like a pretty good idea to me.

**Pull request merged**
